rust/main/agents/relayer/src/msg/metadata/multisig/base.rs

```diff
@@ -105,14 +107,29 @@ pub trait MultisigIsmMetadataBuilder: AsRef<MessageMetadataBuilder> + Send + Syn
     ) -> Result<(Vec<H256>, u8), MetadataBuildError> {
         let ism_domain = multisig_ism.domain().name();
         let fn_key = "validators_and_threshold";
-        // To have the cache key be more succinct, we use the message id
-        let call_params = (multisig_ism.address(), message.id());
+
+        // Depending on the cache policy, make use of the message ID
+        let params_cache_key = match self
+            .as_ref()
+            .base_builder()
+            .ism_cache_policy_classifier()
+            .get_cache_policy(
+                self.as_ref().root_ism,
+                multisig_ism.domain(),
+                self.module_type(),
+            )
+            .await
+        {
+            // To have the cache key be more succinct, we use the message id
+            IsmCachePolicy::MessageSpecific => (multisig_ism.address(), message.id()),
+            IsmCachePolicy::IsmSpecific => (multisig_ism.address(), H256::zero()),
+        };
```

Comment:
using either the (ism_address, message_id) or (ism_address, zero_id) as caching keys seems ISM agnostic and could be refactored out

Reply:
definitely agree the whole caching setup is clunky and could use a refactor - I want us to revisit this later but due to the urgency I'm gonna go ahead with merging